### PR TITLE
Enable GCP Filestore on cluster

### PR DESCRIPTION
**Pull request by pawloch00**
## Fixes / Features
- Enable passing GCPFilestore flag during cluster creation so cluster create has --addons GcpFilestoreCsiDriver flag.
- add GCPFilestore type to storage create command so if GCPFilestoreCsiDriver is not enabled cluster is updated.

## Testing / Documentation
Testing details.

- [ y/n ] Tests pass
- [ y/n ] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: src/xpk/core/core.py [hunk 1]
```diff
@@ -1200,6 +1198,33 @@ def is_gcsfuse_driver_enabled_on_cluster(args) -> bool:
...
+      args,
+  )
+  if return_code != 0:
+    xpk_exit(return_code)
+  if gcpfilestore_driver_enabled.lower() == 'true':
+    xpk_print(
+        'GCPFilestore CSI driver is enabled on the cluster, no update needed.'
+    )
+    return True
+  return False
```
Comment (BluValor) [inline src/xpk/core/core.py:1225]: Code duplication. Extract to separate function similar to update_gke_cluster_with_addon. Possible signature: is_driver_enabled(driver_config_name: str) -> bool.

File: src/xpk/core/core.py [hunk 2]
```diff
@@ -1285,6 +1310,26 @@ def update_cluster_with_gcsfuse_driver_if_necessary(args) -> int:
...
+  Args:
+    args: user provided arguments for running the command.
+  Returns:
+    0 if successful and error code otherwise.
+  """
+
+  if is_gcpfilestore_driver_enabled_on_cluster(args):
+    return 0
+  cluster_update_return_code = update_gke_cluster_with_addon(
+      'GcpFilestoreCsiDriver', args
```
Comment (BluValor) [inline src/xpk/core/core.py:1324]: Same as: https://github.com/AI-Hypercomputer/xpk/pull/215/files#r1842071392

File: .github/workflows/build_tests.yaml [hunk 3]
```diff
@@ -62,7 +62,7 @@ jobs:
       run: |
         python3 xpk.py cluster create-pathways --cluster $TPU_CLUSTER_NAME --tpu-type=v4-8  --num-slices=2 \
         --zone=us-central2-b --default-pool-cpu-machine-type=n1-standard-16 --default-pool-cpu-num-nodes=16 \
-        --reservation='${{ secrets.GCP_TPU_V4_RESERVATION }}' --enable-workload-identity --enable-gcsfuse-csi-driver
+        --reservation='${{ secrets.GCP_TPU_V4_RESERVATION }}' --enable-workload-identity --enable-gcsfuse-csi-driver --custom-cluster-arguments="${CLUSTER_ARGUMENTS}"
```
Comment (BluValor) [inline .github/workflows/build_tests.yaml:65]: Do we not create separate tests for creating cluster with filestore storage or fuse storage? Or do we leave it until later - when we will use them in workloads?

File: src/xpk/core/core.py (hunk 1 above)
Comment (BluValor) [reply src/xpk/core/core.py:1225]: Actually there are more functions similar to these ones, with varying format of the command output and comments printed out. I would suggest creating additional function in this case:
check_clusster_config(format: str, task: str, positive_msg: str, truth_check) -> bool

Please give me your opinion about this as this is a little bit of a bigger refactor, although still contained within one file.

**Review by mbobrovskyi — APPROVED** [review]

File: .github/workflows/build_tests.yaml (hunk 3 above)
Comment (pawloch00) [reply .github/workflows/build_tests.yaml:65]: Running tests with filestore is time consuming. I will leave it as it is

File: src/xpk/core/core.py (hunk 1 above)
Comment (pawloch00) [reply src/xpk/core/core.py:1225]: I will refactor to is_driver_enabled please create refactoring task for other places

**Review by BluValor — APPROVED** [review]

**Pull request merged**